@here/harp-mapview/test/SolidLineMeshTest.ts

```diff
@@ -0,0 +1,517 @@
+        ).build();
+        const localToWorld = new THREE.Matrix4().getInverse(worldToLocal);
+        mesh.applyMatrix4(localToWorld);
+        mesh.updateMatrixWorld();
+
+        const intersects: THREE.Intersection[] = [];
+        const { raycaster, distance } = buildRayTo([1, 0, 0], [1, 0, 1]);
+        mesh.raycast(raycaster, intersects);
+
+        checkIntersect(intersects, {
+            distance,
+            point: new THREE.Vector3(1, 0, 0),
+            index: 0,
+            object: mesh
+        });
+    });
+
+    it("raycast returns intersection on hit with mesh with multiple features", function() {
+        const mesh = new SolidLineMeshBuilder([0, 0, 0, 1, 0, 0, 2, 0, 0], [0, 1, 0])
+            .addFeature([5, 5, 5, 7, 7, 7], [-1, 1, 1])
+            .build();
+        const intersects: THREE.Intersection[] = [];
+        const { raycaster, distance } = buildRayTo([5, 5, 5], [5, 5, 7]);
+        mesh.raycast(raycaster, intersects);
+
+        expect(mesh.userData.feature?.boundingVolumes)
+            .is.an("array")
+            .with.lengthOf(2);
+        expect(mesh.userData.feature.boundingVolumes[0]).deep.equals(
+            new THREE.Sphere(new THREE.Vector3(1, 0, 0), 1)
+        );
+        expect(mesh.userData.feature.boundingVolumes[1]).deep.equals(
+            new THREE.Sphere(new THREE.Vector3(6, 6, 6), Math.sqrt(3))
+        );
+        checkIntersect(intersects, {
+            distance,
+            point: new THREE.Vector3(5, 5, 5),
+            index: 12,
+            object: mesh
+        });
+    });
+
+    it("raycast returns intersection on hit with mesh with multiple groups", function() {
+        const mesh = new SolidLineMeshBuilder([0, 0, 0, 1, 0, 0, 2, 0, 0], [0, 1, 0])
+            .addFeature([5, 5, 5, 7, 7, 7], [-1, 1, 1])
+            .addGroup(0, 1)
+            .addGroup(1, 2, new SolidLineMaterial({ lineWidth: 5 }))
+            .build();
+        const intersects: THREE.Intersection[] = [];
+        const { raycaster, distance } = buildRayTo([5, 5, 5], [5, 5, 7]);
+        mesh.raycast(raycaster, intersects);
+
+        checkIntersect(intersects, {
+            distance,
+            point: new THREE.Vector3(5, 5, 5),
+            index: 12,
+            object: mesh
+        });
+    });
+
+    it("raycast on displaced geometry expands bounding sphere by displacement range", function() {
+        const meshBuilder = new SolidLineMeshBuilder([0, 0, 0, 2, 0, 0], [0, 1, 0]);
+        meshBuilder.geometryBuilder.withUv([0, 0, 0, 0]).withNormal([0, 0, 1, 0, 0, 1]);
+        const mesh = meshBuilder.build();
+        mesh.geometry = new DisplacedBufferGeometry(
+            mesh.geometry as THREE.BufferGeometry,
+            new THREE.DataTexture(new Float32Array([0]), 1, 1),
+            { min: 10, max: 12 }
+        );
+        const intersects: THREE.Intersection[] = [];
+        const { raycaster } = buildRayTo([0, 1, 0]);
+        mesh.raycast(raycaster, intersects);
```

Comment:
Could we check for intersection with displaced mesh, I haven't found any test case for that or I missed something?

Reply:
done.